src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1300,6 +1300,20 @@ var ReactCompositeComponent = {
     return inst;
   },
 
+  toTree: function() {
+    const element = this._currentElement;
+    // not using `children`, but I don't want to rewrite without destructuring
+    // eslint-disable-next-line no-unused-vars
+    const { children, ...propsWithoutChildren } = element.props;
+    return {
+      nodeType: ReactNodeTypes.getType(element),
+      type: element.type,
+      props: propsWithoutChildren,
+      instance: this._instance,
+      rendered: this._renderedComponent.toTree(),
+    };
+  },
```

Comment:
Historically we've avoided adding/using test-renderer specific APIs to `ReactCompositeComponent` (reference: https://github.com/facebook/react/pull/7649#r78279440) Could we derive this tree from the instance inside the stack test renderer like you did with Fiber?

Maybe it could be abstracted into a module like `ReactInstanceToTree` or something like that?